doc/ioc/tutorials/streamdevice.md

```diff
@@ -0,0 +1,288 @@
+---
+title: "Creating a StreamDevice IOC"
+---
+
+In this tutorial,
+you're gonna learn how to create an EPICS IOC with EPNix
+that communicates with a power supply,
+using the [StreamDevice] support module.
+
+  [StreamDevice]: https://paulscherrerinstitute.github.io/StreamDevice/
+
+# Pre-requisites
+
+Verify that you have all pre-requisites installed.
+If not,
+follow the [Pre-requisites] section.
+
+  [Pre-requisites]: ./pre-requisites.md
+
+# Running the power supply simulator
+
+EPNix has a power supply simulator
+for you to test your IOC.
+
+To run it:
+
+``` bash
+nix run 'github:epics-extensions/epnix#psu-simulator'
+```
+
+For the rest of the tutorial,
+leave it running in a separate terminal.
+
+# Creating your top
+
+We can use these command to create an EPNix top:
+
+``` bash
+# Initialise an EPNix top
+nix flake new -t 'github:epics-extensions/epnix' my-top
+cd my-top
+
+# Enter the EPNix development shell, that has EPICS base installed in it.
+nix develop
+
+# Create your app and ioc boot folder
+makeBaseApp.pl -t ioc example
+makeBaseApp.pl -i -t ioc -p example -a linux-x86_64 Example
+
+# Create a git repository, and make sure all files are tracked
+git init
+git add .
```

Comment:
According to @agaget, this needs to be before `nix develop`. Not sure why.

If this should be before `nix develop`, we'll need to re-add files after `makeBaseApp.pl`.

Reply:
according to me and my new test it doesn't seem to be a problem anymore. I will add more logs in my issues in the future ....